Classes/PlusOnes.lua

```diff
@@ -2,126 +2,970 @@
+        and uuid == DB:get('PlusOnes.MetaData.uuid', '')
+        and lastUpdate >= DB:get('PlusOnes.MetaData.updatedAt', 0)
+    ) then
+        return;
+    end
+
+    -- Looks like you need my data, here it is!
+    GL.CommMessage.new(
+        CommActions.broadcastPlusOnesData,
+        {
+            importString = self:export(false),
+            MetaData = DB:get("PlusOnes.MetaData", {}),
+        },
+        "WHISPER",
+        CommMessage.Sender.name
+    ):send();
+end
+
+--- Add points to a give user's balance
 ---
 ---@param playerName string
----@return number
-function PlusOnes:get(playerName)
-    GL:debug("PlusOnes:get");
+function PlusOnes:addPlusOnes(playerName)
+    GL:debug("PlusOnes:addPlusOnes")
 
-    playerName = GL:normalizedName(playerName);
+    local currentPoints = self:getPlusOnes(playerName) or 0;
+    self:queueUpdate(playerName, currentPoints + 1);
+end
 
-    return DB.PlusOnes[playerName] or 0;
+--- Subtract points from a given user's balance
+---
+---@param playerName string
+function PlusOnes:subtractPlusOnes(playerName)
+    GL:debug("PlusOnes:subtractPlusOnes");
+
+    local currentPoints = self:getPlusOnes(playerName) or 0;
+
+    self:queueUpdate(playerName, max(currentPoints - 1, 0));
 end
 
---- Set a player's PlusOnes value
+--- Queue an update until broadcast is finished
 ---
 ---@param playerName string
----@param value number
----@return void
-function PlusOnes:set(playerName, value)
-    GL:debug("PlusOnes:set");
+---@param plusOne number
+---@param delete boolean
+function PlusOnes:queueUpdate(playerName, plusOne, delete)
+    local dontBroadcast = true;
 
-    -- A table was provided, treat it as a mass assignment
-    if (type(playerName) == "table") then
-        return self:massSet(playerName);
+    if (plusOne) then
+        self:setPlusOnes(playerName, plusOne, dontBroadcast);
     end
 
-    playerName = GL:normalizedName(playerName);
+    if (delete) then
+        self:deletePlusOnes(playerName, dontBroadcast);
+    end
+
+    local Update = {
+        playerName = playerName,
+        plusOne = plusOne or nil,
+        delete = delete or false,
+    };
+
+    tinsert(self.QueuedUpdates, Update);
 
-    DB.PlusOnes[playerName] = GL:round(value or 0);
+    -- Fire an event to let the application know that an update was queued
+    GL.Events:fire("GL.PLUSONES_UPDATE_QUEUED");
 end
 
---- Assign PlusOne values en masse
+--- Send out the queued updates
 ---
----@param plusOnesByPlayerName table
 ---@return void
-function PlusOnes:massSet(plusOnesByPlayerName)
-    GL:debug("PlusOnes:massSet");
+function PlusOnes:broadcastQueuedUpdates()
+    GL:debug("PlusOnes:broadcastQueuedUpdates");
 
-    for playerName, value in pairs(plusOnesByPlayerName) do
-        playerName = GL:normalizedName(playerName or 0);
+    if (not GL.User.isInGroup) then
+        return false;
+    end
 
-        DB.PlusOnes[playerName] = GL:round(value or 0);
+    --if (not self:userIsAllowedToBroadcast()) then
```

Comment:
You sure we don't want to put this in place?

Reply:
PlusOnes database is now similar to Boosted Rolls.  The PlusOnes data is accessed as DB.PlusOnes.Totals[playerName].total